verl/trainer/ppo/ray_trainer.py

```diff
@@ -919,6 +939,18 @@ def fit(self):
 
                         batch.batch['token_level_scores'] = reward_tensor
 
+                        if self.config.trainer.log_rollout_generations > 0:
+                            # Extract inputs, outputs and scores from the batch
+                            input_ids = batch.batch['input_ids']
```

Comment:
can we move the preprocessing logic for logging to the _log_all_generations function instead. also rename `self._log_all_generations` to `self._maybe_log_all_generations` and move the condition check `if self.config.trainer.log_rollout_generations` inside the function. 

In principle we want to keep the training loop succinct